src/taco-cli/cli/utils/cordovaWrapper.ts

```diff
@@ -170,10 +170,22 @@ class CordovaWrapper {
         });
     }
 
-    public static run(commandData: commands.ICommandData, platforms: string[] = null): Q.Promise<any> {
-        return CordovaWrapper.cordovaApiOrProcess((cordova: Cordova.ICordova) => {
-            return cordova.raw.run(CordovaHelper.toCordovaRunArguments(commandData, platforms));
-        }, () => ["run"].concat(CordovaHelper.toCordovaCliArguments(commandData, platforms)));
+    public static run(commandData: commands.ICommandData, platforms: string[] = null): Q.Promise <any> {
+	return CordovaWrapper.cordovaApiOrProcess((cordova: Cordova.ICordova) => {
+            // First install the plugin if it's not yet installed on this project
+            return cordova.raw.plugin('ls', null, null).then(function(pluginList: string[]): Q.Promise <any> {
```

Comment:
This is going to always add the plugin, regardless of whether livereload or devicesync are specified. Perhaps it'd be better to move the plugin adding logic to the calling code? Or at least add a check for whether the flags are specified.